gensim/corpora/utils.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Author: Gensim Contributors
+# Copyright (C) 2020 RaRe Technologies s.r.o.
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+import logging
+import os
+import itertools
+from collections import namedtuple
+
+from gensim import utils
+
+try:
+    from gensim.models.word2vec_inner import MAX_WORDS_IN_BATCH
```

Comment:
There isn't truly any dependency of these utility classes on some other-module Cython optimizations. sp this seems off in terms of direction-of-dependency, fragility, exception raised, etc.

Perhaps instead: a reference value is defined with a comment here, then used both below *and* imported from here into the other modules that want to be in-sync on this arbitrary cap? EG: 

```
#: Shared capped size, in count of words/tokens, for texts that should not be arbitrarily long
MAX_WORDS=10000
```